cfme/tests/cloud/test_providers.py

```diff
@@ -22,20 +23,163 @@
+def az_pwsh_vm(appliance, request):
+    """
+    azure_pwsh contains powershell and necessary scripts to upload vhd, create VM, get ip of the
+    resource and delete the VM.
+    Find the provider that contains that template.
+    TODO(anikifor): define the configuration of that VM as cloud-init file
+    """
+    providers_data = cfme_data.get("management_systems", {})
+
+    for provider, values in providers_data.items():
+        templates = values.get('templates')
+        if templates:
+            az_template = templates.get('powershell_vm')
+            if az_template:
+                provider_key = provider
+
+    if not provider_key:
+        pytest.skip("There's no provider that contains a template with powershell")
+    provider = get_crud_by_name(provider_key)
+
+    vm_name = random_vm_name(context="pwsh")
+    pwsh_vm = provider.data.templates.powershell_vm.name
+
+    collection = provider.appliance.provider_based_collection(provider)
+
+    try:
+        vm = collection.instantiate(vm_name, provider, pwsh_vm)
+        vm.create_on_provider(allow_skip="default")
+    except IndexError:
+        from cfme.exceptions import ItemNotFound
+        raise ItemNotFound('VM with powershell not found!')
+
+    vm.mgmt.ensure_state(VmState.RUNNING)
+
+    @wait_for_decorator(timeout="10m", delay=5)
+    def get_ip_address():
+        ip = vm.mgmt.ip
+        return ip is not None
+    yield vm
+
+    vm.cleanup_on_provider()
+
+
+@pytest.fixture
+def pwsh_ssh(az_pwsh_vm):
+    """Provide vm_ssh_client for ssh operations in the test."""
+    with ssh.SSHClient(hostname=az_pwsh_vm.ip_address,
+                       username=credentials['host_default']['username'],
+                       password=credentials['host_default']['password']) as vm_ssh_client:
+        yield vm_ssh_client
+
+
+@pytest.fixture
+def connect_az_account(pwsh_ssh):
+    """
+    Connect to Azure account to run further scripts, see
+    https://docs.microsoft.com/en-us/powershell/azure/authenticate-azureps
+    """
+    pwsh_ssh.run_command("{} pwsh connect_account.ps1".format(CD_SPACE), timeout=180)
+
+
+@pytest.fixture(scope='function')
+def cfme_vhd(appliance, pwsh_ssh):
+    ip_of_recourse = pwsh_ssh.run_command(
+        '{} pwsh get_ip.ps1 | grep -oE "([0-9]{{1,3}}\.){{3}}[0-9]{{1,3}}"'.format(CD_SPACE),
+        timeout=60).output.strip()
+    if ip_of_recourse is not None:
+        pytest.skip("The resource is taken by some other VM in Azure")
+    stream = appliance.version.stream()
+    try:
+        url = conf.cfme_data['basic_info']['cfme_images_url'][stream]
+    except KeyError:
+        pytest.skip("Skipping since no such key found in yaml")
+
+    image = pwsh_ssh.run_command(
+        """wget -qO- {url} | grep -Po '(?<=href=")[^"]*' | grep azure""".format(url=url),
+        timeout=30).output.strip()
+    image_url = '{}/{}'.format(url, image)
+    pwsh_ssh.run_command("{cd} wget {image}".format(cd=CD_SPACE, image=image_url), timeout=180)
```

Comment:
@izapolsk  can you explain the benefits? Of course, I know the arguments but to me the current way is more clear.

Reply:
I think I agree with @izapolsk here, it would be better to use such options (and in other commands provide direct paths).  There are no assertions against the ssh commands made below, and if the `cd` fails for some reason, all the subsequent commands will fail.

To me its more clear to provide full paths to the files we're manipulating, and you should be able to construct those with the constant.

Reply:
@mshriver ok, I'm changing it

Reply:
@lina-nikiforova, there are several reasons why I'd better see using arguments there.
 main reason is that you rely on default behavior of tools which can be easily changed and may vary from system to system or between versions. so, you need to exactly specify what you want this tool to do.